main.tf

```
@@ -0,0 +1,121 @@
+data "template_file" "master_0" {
+  count = "1"
+
+  template = "${file("${path.module}/templates/init.yaml")}"
+
+  vars {
+    # Kubernetes.
+
+    kubernetes_ca_crt      = "${base64encode(var.kubernetes_ca_crt)}"
+    kubernetes_ca_key      = "${base64encode(var.kubernetes_ca_key)}"
+    token                  = "${var.kubernetes_token}"
+    api_server_cert_sans   = "${join(", ", var.master_ip_addresses)}"
+    api_endpoint           = "${var.master_ip_addresses[0]}"
+    control_plane_endpoint = "${var.master_ip_addresses[0]}"
+    taints                 = ""
+    labels                 = ""
+    cluster_name           = "${var.cluster_name}"
+    dns_domain             = "${var.dns_domain}"
+    pod_subnet             = "${var.pod_subnet}"
+    service_subnet         = "${var.service_subnet}"
+
+    # Talos.
+
+    os_ca_crt                   = "${base64encode(var.talos_ca_crt)}"
+    os_ca_key                   = "${base64encode(var.talos_ca_key)}"
+    trustd_username             = "${var.trustd_username}"
+    trustd_password             = "${var.trustd_password}"
+    trustd_endpoints            = "[${var.master_ip_addresses[1]}]"
```

```suggestion
    trustd_endpoints            = "[${element(var.master_hostnames, count.index + 1)}]"
```
I use the `element` function here as Terraform interpolation will automatically wrap when `count.index + 1` goes out-of-bounds.